txscript/script.go

```diff
@@ -420,6 +446,216 @@ func CalcWitnessSigHash(script []byte, sigHashes *TxSigHashes, hType SigHashType
 	return calcWitnessSignatureHashRaw(script, sigHashes, hType, tx, idx, amt)
 }
 
+// sigHashExtFlag represents the sig hash extension flag as defined in BIP 341.
+// Extensions to the base sighash algorithm will be appended to the base
+// sighash digest.
+type sigHashExtFlag uint8
+
+const (
+	// baseSigHashExtFlag is the base extension flag. This adds no changes
+	// to the sighash digest message. This is used for segwit v1 spends,
+	// a.k.a the tapscript keyspend path.
+	baseSigHashExtFlag sigHashExtFlag = 0
+)
+
+// taprootSigHashOptions houses a set of functional options that may optionally
+// modify how the taproot/script sighash digest algorithm is implemented.
+type taprootSigHashOptions struct {
+}
+// defaultTaprootSighashOptions returns the set of default sighash options for
+// taproot execution.
+func defaultTaprootSighashOptions() *taprootSigHashOptions {
+	return &taprootSigHashOptions{}
+}
+
+// TaprootSigHashOption defines a set of functional param options that can be
+// used to modify the base sighash message with optional extensions.
+type TaprootSigHashOption func(*taprootSigHashOptions)
+// calcTaprootSignatureHashRaw computes the sighash as specified in BIP 143.
+// If an invalid sighash type is passed in, an error is returned.
+func calcTaprootSignatureHashRaw(sigHashes *TxSigHashes, hType SigHashType,
+	tx *wire.MsgTx, idx int,
+	prevOutFetcher PrevOutputFetcher,
+	sigHashOpts ...TaprootSigHashOption) ([]byte, error) {
+
+	opts := defaultTaprootSighashOptions()
+	for _, sigHashOpt := range sigHashOpts {
+		sigHashOpt(opts)
+	}
+
+	// If a valid sighash type isn't passed in, then we'll exit early.
+	if !isValidTaprootSigHash(hType) {
+		// TODO(roasbeef): use actual errr here
```

Comment:
Is this TODO still relevant? Since there is no `err` returned by `isValidTaprootSigHash`?